app/controllers/admin/cases_controller.rb

```diff
@@ -97,7 +87,43 @@ def update
         @case.publisher.status_updates.create(status: PublisherStatusUpdate::ACTIVE, publisher_note: note)
       end
 
-      redirect_to admin_cases_path
+      return redirect_to [:admin, @case], flash: { notice: "Case has been moved back to in progress"} if params[:status] == Case::IN_PROGRESS
+
+      next_case = Case.where(assignee: current_user, status: Case::IN_PROGRESS).take
+      if next_case
+        redirect_to [:admin, next_case], flash: { notice: "Previous case #{@case.number} has been marked #{params[:status]}"}
+      else
+        redirect_to admin_cases_path
+      end
+    end
+
+    private
+
+    def parse_search(search_case, queries)
+      statuses = queries.select { |x| x.include?("status") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+      assigned = queries.select { |x| x.include?("assigned") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+
+      first_status = statuses[0]
+      if first_status.present?
+        value = first_status
+        search_case = search_case.where(status: value)
+
+        statuses[1..-1].each do |value|
```

Comment:
Block 111 to 114 can be refactored to 

```
search_case = search_case.where(status: statuses)
```

Reply:
That would be an `AND` statement, rather than an `OR` which would result in no results. 😅

Reply:
`statuses` is an array of `Case.status` right?

```
2.5.7 :006 > Case.where(status: ["closed", "open"]).to_sql
 => "SELECT \"cases\".* FROM \"cases\" WHERE \"cases\".\"status\" IN ('closed', 'open')"
Case.where(status: ["closed", "open"]).count
=> 548 // proves that it's not an AND
```

Reply:
Wow! that's awesome, had no idea.

https://github.com/brave-intl/publishers/pull/2627/commits/b4dec27a671494c732b67884ec1db5aa1767ef27